### PR TITLE
Storybook: add eslint plugin & fix lint problems

### PR DESCRIPTION
**Pull request by julianiff**
Hey, i added the eslint checking to vite this time. Now we get updates while working in storybook. I put everything on warning to not break the build and will contact someone who works on the richtext, as there are some more errors that need to be looked at.

### CODE REVIEW

File: .storybook/main.ts [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
     },
     stories: ['../src/**/*.stories.tsx'],
     addons: ['storybook-dark-mode', '@storybook/addon-links', '@storybook/addon-essentials', '@storybook/addon-a11y'],
     staticDirs: ['assets'],
-    features: {
-        storyStoreV7: true,
-        buildStoriesJson: true,
-    },
+    // features: {
+    //     storyStoreV7: true,
```
Comment (julianiff) [inline .storybook/main.ts:16]: i removed storyStore again, cause doesnt work super great

File: src/typography/shared/Shared.stories.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
+export default {};
```
Comment (SamuelAlev) [inline src/typography/shared/Shared.stories.tsx:3]: File can be renamed to `shared.ts` no?
Then the `export default` would no longer be required

Comment (julianiff) [reply src/typography/shared/Shared.stories.tsx:3]: thx for the input, changing it

File: .storybook/main.ts [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
+    // },
```
Comment (julianiff) [inline .storybook/main.ts:17]:
```suggestion
```

File: .storybook/main.ts (hunk 1 above)
Comment (julianiff) [reply .storybook/main.ts]: you are right

**Review by imoutaharik — APPROVED** [review]

**Pull request merged**